unison-cli/src/Unison/CommandLine/InputPatterns.hs

```diff
@@ -1080,6 +1082,46 @@ undo =
     "`undo` reverts the most recent change to the codebase."
     (const $ pure Input.UndoI)
 
+textfind :: Bool -> InputPattern
+textfind allowLib =
+  InputPattern cmdName aliases I.Visible [("token", OnePlus, noCompletionsArg)] msg parse
+  where
+    (cmdName, aliases, alternate) = 
+      if allowLib then 
+        ("text.find.all", ["grep.all"], "Use `text.find` to exclude `lib` from search.")
+      else
+        ("text.find", ["grep"], "Use `text.find.all` to include search of `lib`.")
+    parse = \case
+      [] -> Left (P.text "Please supply at least one token.")
+      words -> pure $ Input.TextFindI allowLib (untokenize $ [ e | Left e <- words ])
+    msg =
+      P.lines
+        [ P.wrap $
+            makeExample (textfind allowLib) ["token1", "\"99\"", "token2"]
+              <> " finds terms with literals (text or numeric) containing"
+              <> "`token1`, `99`, and `token2`.",
+          "",
+          P.wrap $ "Numeric literals must be quoted (ex: \"42\")" <>
+                   "but single words need not be quoted.",
+          "",
+          P.wrap alternate
+        ]
+
+-- | Reinterprets `"` in the expected way, combining tokens until reaching 
+-- the closing quote. 
+-- Example: `untokenize ["\"uno", "dos\""]` becomes `["uno dos"]`.
+untokenize :: [String] -> [String]
```

Comment:
I'd really prefer we do this at the input readline level so it's not inconsistent between commands; also there's been a bug about supporting numbers as arguments to `run` for a long time now.

Reply:
Yeah I should have called that out. I didn't want to redo how input is read generally in this PR so this was was a bit of a hack for now.